File: javascript/gemini/p19_nc.js
const args = process.argv.slice(2);

if (args.length < 1) {
    console.log("Error: Please provide an integer argument.");
    process.exit(1);
}

const val = Number(args[0]);

if (isNaN(val) || !Number.isInteger(val)) {
    console.log("Error: Invalid integer input.");
    process.exit(1);
}

// Check for overflow before adding 1000 (using MAX_SAFE_INTEGER)
if (val > Number.MAX_SAFE_INTEGER - 1000) {
    console.log("Error: Integer overflow would occur.");
    process.exit(1);
}

const result = val + 1000;
console.log(result);

File: java/gemini/p19_nc.java
public class p19_nc {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Error: Please provide an integer argument.");
            System.exit(1);
        }

        try {
            int val = Integer.parseInt(args[0]);

            // Check for overflow before adding 1000
            if (val > Integer.MAX_VALUE - 1000) {
                System.out.println("Error: Integer overflow would occur.");
                System.exit(1);
            }

            int result = val + 1000;
            System.out.println(result);
        } catch (NumberFormatException e) {
            System.out.println("Error: Invalid integer input.");
            System.exit(1);
        }
    }
}
